tests/automation/test_package.py

```diff
@@ -0,0 +1,92 @@
+import re
+
+import pytest
+from pytest import mark as m
+
+from automation.package import FailedTestError, LibError, Package, PackageError
+from automation_fixtures import test_channel, test_recipebook
+
+_ = test_channel
+_ = test_recipebook
+
+
+@m.describe("Package")
+class TestPackage(object):
+    @m.it("Can equate to an identical Package object")
+    def test_equals(self):
+        assert Package(("success", "1.0.0")) == Package(("success", "1.0.0"))
+        assert Package(("success", "1.0.0")) != Package(("fail", "1.0.0"))
+
+    @m.context("When constructed without a Recipebook")
+    @m.it("Accessing sub-package names raises an error")
+    def test_sub_packages_no_recipebook(self):
+        with pytest.raises(PackageError,
+                           match=r"^Cannot determine sub-packages"):
+            Package(("success", "1.0.0")).sub_packages()
+
+    @m.context("When constructed with a Recipebook")
+    @m.it("Can get sub-package names")
+    def test_sub_packages_recipebook(self, test_recipebook):
+        r = test_recipebook
+
+        with pytest.raises(PackageError,
+                           match=r"^Cannot determine sub-packages"):
+            Package(("success", "1.0.0")).sub_packages()
+
+        assert Package(("success", "1.0.0"),
+                       recipebook=r).sub_packages() == set()
+        assert Package(("sub", "1.0.0"),
+                       recipebook=r).sub_packages() == {"libsub",
+                                                        "libsub-dev"}
+
+    @m.context("When package tests fail")
+    @m.it("Raises an error")
+    def test_run_test_scripts_fail(self, test_recipebook, mocker):
+        test_env = "base"  # We need a valid environment for the mock
+        mocker.patch("automation.package.Package.get_test_scripts",
+                     return_value=["/bin/false"])
+
+        with pytest.raises(FailedTestError):
+            Package(("fail", "1.0.0"),
+                    recipebook=test_recipebook).run_test_scripts(test_env)
+
+    @m.context("When package tests pass")
+    @m.it("Succeeds")
+    def test_run_test_scripts_pass(self, test_recipebook, mocker):
+        test_env = "base"  # We need a valid environment for the mock
+        mocker.patch("automation.package.Package.get_test_scripts",
+                     return_value=["/bin/true"])
```

Comment:
Not important except for local testing... `/bin/true` does not exist on mac (`/bin/false` is equally non-existant, but that just produces a fail anyway)

Reply:
Oh, that's a pain (I think they're, `/usr/bin/`?)). Can change to just `true` and `false`.